dataset_interfaces/interface.py

```diff
@@ -187,12 +189,56 @@ def load(cls, file_path: Path | str) -> "TestExample":
         d["waits"] = [WaitCreator.unserialise(x) for x in d["waits"]]
         d["example_id"] = file_path.name.removesuffix(".def.json")
         d["dataset_name"] = file_path.parent.name
-        assert d["dataset_name"] in DATASETS_BY_NAME, f"Couldn't find a generator for dataset {d['dataset_name']}."
-        dataset_generator = DATASETS_BY_NAME[d["dataset_name"]]()
         generator_attrs = {"dataset_name", "description", "reset_message"}
         d = {k: d[k] for k in d.keys() if k not in generator_attrs}
         return dataset_generator.create_example(**d)
 
+    def default_waits(self):
+
+        if isinstance(self, DynamicExample):
+            return
+
+        assert len(self.is_question) >= 1, "There are no questions for this test"
+        assert len(self.waits) == 0 or len(self.waits) == len(self.is_question), "Current waits should be empty or the same length as the script"
+        memory_span = self.dataset_generator.memory_span
+
+        # TODO: All this token counting is based on OpenAIs methods, which is not Anthropics - but Aanthopic has no reliable tokeniser released so....
+        script_tokens = get_tokens_for_script(self.script)
+        assert script_tokens < memory_span, f"The script for test {self.dataset_name} is too long (estimated {script_tokens} tokens) for the specified memory span of {memory_span} tokens."
+
+        span_minus_script = memory_span - script_tokens
+        needle_question_proportions = (.75, .25) if self.number_of_questions == 1 else (.5, .5)
+
+        # Figure out what our spacing is going to be. We will do the 75%/25% needle distribution.
+        tokens_for_needles = math.floor(span_minus_script * needle_question_proportions[0])
+        needles = len(self.is_question) - sum(self.is_question)
+        token_wait_per_needle = tokens_for_needles // needles
+        token_wait_between_questions = math.floor(span_minus_script * .9 * needle_question_proportions[1]) // sum(self.is_question)
```

Comment:
I thought you were going to spread the script lines across the memory span (or 90% of it).

Reply:
I thought I wasn't? I said that it worked fine as-is for 8k tokens, and with the warnings and guards we could just leave it for now and see if there is an issue.

Reply:
Alright then. It just seems odd to me that those proportions are so arbitrary. I think I'd like it more if `span_minus_script * 0.9` would be split in equal pieces and placed between every two consecutive script lines.

Reply:
Thinking more on it, it is easier to explain that we just uniformly distribute everything over the 90% of the span. So I have changed it for that.